mne/io/base.py

```diff
@@ -1426,9 +1427,19 @@ def save(self, fname, picks=None, tmin=0, tmax=None, buffer_size_sec=None,
         endings += tuple([f'{e}.gz' for e in endings])
         endings_err = ('.fif', '.fif.gz')
 
-        # convert to str, check for overwrite a few lines later
-        fname = _check_fname(fname, overwrite=True)
-        check_fname(fname, 'raw', endings, endings_err=endings_err)
+        # Expand `~` and convert to a string
+        # Don't use _check_fname() here, as it will be called inside
+        # _write_raw() and we want to avoid repeated logging in case of
+        # overwriting
+        fname = str(Path(fname).expanduser())
+
+        # check naming convention
+        check_fname(
+            fname=fname,
+            filetype='raw',
+            endings=endings,
+            endings_err=endings_err
+        )
```

Comment:
This code is less DRY, but also less functional as it won't _validate_type as path-like (right)? Instead can you just do:
`_check_fname(fname, overwrite=True, verbose='error')`? This won't log anything

Reply:
`_check_fname` currently doesn't have a `verbose` param, but could make sense to add one.

@sappelhoff, wanna take over? ;) I'll be busy with other things until tomorrow or so